packages/ee/lib/service.ts

```diff
@@ -89,47 +89,86 @@ const fetchLicenseForE2ETesting = async (): Promise<{
   }
 };
 
-export const getIsEnterpriseEdition = async (): Promise<boolean> => {
+export const getEnterpriseLicense = async (): Promise<{
+  active: boolean;
+  features: TEnterpriseLicenseFeatures | null;
+  lastChecked: Date;
+  isPendingDowngrade?: boolean;
+}> => {
   if (!ENTERPRISE_LICENSE_KEY || ENTERPRISE_LICENSE_KEY.length === 0) {
-    return false;
+    return {
+      active: false,
+      features: null,
+      lastChecked: new Date(),
+    };
   }
 
   if (E2E_TESTING) {
     const previousResult = await fetchLicenseForE2ETesting();
-    return previousResult && previousResult.active !== null ? previousResult.active : false;
+    // return previousResult && previousResult.active !== null ? previousResult.active : false;
+    return {
+      active: previousResult ? previousResult.active : false,
+      features: previousResult ? previousResult.features : null,
+      lastChecked: previousResult ? previousResult.lastChecked : new Date(),
+    };
   }
 
   // if the server responds with a boolean, we return it
   // if the server errors, we return null
   // null signifies an error
   const license = await fetchLicense();
+
   const isValid = license ? license.status === "active" : null;
+  const threeDaysInMillis = 3 * 24 * 60 * 60 * 1000;
+  const currentTime = new Date();
 
   const previousResult = await getPreviousResult();
+
   if (previousResult.active === null) {
     if (isValid === null) {
-      await setPreviousResult({
+      const newResult = {
         active: false,
         features: { isMultiOrgEnabled: false },
         lastChecked: new Date(),
-      });
-      return false;
+      };
+
+      await setPreviousResult(newResult);
+      return newResult;
     }
   }
 
   if (isValid !== null && license) {
-    await setPreviousResult({ active: isValid, features: license.features, lastChecked: new Date() });
-    return isValid;
+    const newResult = {
+      active: isValid,
+      features: license.features,
+      lastChecked: new Date(),
+    };
+
+    await setPreviousResult(newResult);
+    return newResult;
   } else {
     // if result is undefined -> error
     // if the last check was less than 72 hours, return the previous value:
-    if (new Date().getTime() - previousResult.lastChecked.getTime() <= 3 * 24 * 60 * 60 * 1000) {
-      return previousResult.active !== null ? previousResult.active : false;
+
+    const elapsedTime = currentTime.getTime() - previousResult.lastChecked.getTime();
+    if (elapsedTime < threeDaysInMillis) {
+      return {
+        active: previousResult.active !== null ? previousResult.active : false,
+        features: previousResult.features,
+        lastChecked: previousResult.lastChecked,
+        isPendingDowngrade: true,
+      };
     }
```

Comment:
Can we rearrange the logic here? Should we check for the active license first and fall back to the previous result only if the current one fails?